src/GitHub.Exports.Reactive/Collections/TrackingCollection.cs

```diff
@@ -0,0 +1,472 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Reactive.Linq;
+using System.Reactive.Concurrency;
+using System.Reactive.Disposables;
+using GitHub.Helpers;
+using System.Reactive;
+using System.Linq;
+using ReactiveUI;
+
+namespace GitHub.Collections
+{
+    public class TrackingCollection<T> : ObservableCollection<T>, ITrackingCollection<T>
+        where T : ICopyable<T>
+    {
+        enum TheAction
+        {
+            None,
+            Move,
+            Add,
+            Insert
+        }
+
+        CompositeDisposable disposables = new CompositeDisposable();
+        IObservable<Unit> source;
+        Func<T, T, int> comparer;
+        Func<T, int, IList<T>, bool> filter;
+        IScheduler scheduler;
+        List<T> original;
+
+        public TrackingCollection(Func<T, T, int> comparer = null, Func<T, int, IList<T>, bool> filter = null, IScheduler scheduler = null)
+        {
+            this.scheduler = scheduler ?? RxApp.MainThreadScheduler;
+            this.comparer = comparer;
+            this.filter = filter;
+            if (filter != null)
+                original = new List<T>();
+        }
+
+        public TrackingCollection(IObservable<T> source, Func<T, T, int> comparer = null, Func<T, int, IList<T>, bool> filter = null, IScheduler scheduler = null)
+            : this(comparer, filter, scheduler)
+        {
+            Listen(source);
+        }
+
+        public ITrackingCollection<T> Listen(IObservable<T> obs)
+        {
+            source = obs
+                .ObserveOn(filter != null ? TaskPoolScheduler.Default : scheduler)
```

Comment:
Would it make sense to extract this inner list into our own sorted list? Perhaps `NotifyingSortedList` with a single method for merging items that returns information about what happened?

That'd be a nice way of keeping that responsibility separate from this class.